src/internal/adapter/cast/THEOplayerNativeChromecast.ts

```diff
@@ -1,30 +1,58 @@
-import type { CastState, Chromecast, THEOplayerView } from 'react-native-theoplayer';
+import type { CastEvent, CastState, Chromecast, THEOplayerView } from 'react-native-theoplayer';
+import { CastEventType, PlayerEventType, THEOplayer } from 'react-native-theoplayer';
 import { NativeModules } from 'react-native';
 
 export class THEOplayerNativeChromecast implements Chromecast {
-  public constructor(private readonly _player: THEOplayerView) {}
+  private readonly _player: THEOplayer;
+  private readonly _view: THEOplayerView;
 
-  casting(): Promise<boolean> {
-    return NativeModules.CastModule.chromecastCasting(this._player.nativeHandle);
+  private _casting = false;
+  private _state: CastState = 'available';
+
+  public constructor(player: THEOplayer, view: THEOplayerView) {
+    this._player = player;
+    this._view = view;
+    this._player.addEventListener(PlayerEventType.CAST_EVENT, this._onCastStateChange);
+    void this.init_();
+  }
+
+  async init_(): Promise<void> {
+    this._casting = await NativeModules.CastModule.chromecastCasting(this._view.nativeHandle);
+    this._state = await NativeModules.CastModule.chromecastState(this._view.nativeHandle);
+  }
+
+  private readonly _onCastStateChange = (event: CastEvent) => {
+    if (event.subType === CastEventType.CHROMECAST_STATE_CHANGE) {
+      this._state = event.state;
+      this._casting = event.state === 'connected';
```

Comment:
Just a question: When do we assume that we are 'casting'? Once the state is 'connected' or already when the state is 'connecting'?